src/main/scala/frdomain/ch5/free/AccountRepoMutable.scala

```diff
@@ -35,6 +36,39 @@ case class AccountRepoMutableInterpreter() extends AccountRepoInterpreter {
   def apply[A](action: AccountRepo[A]): Task[A] = action.runM(step)
 }
 
+object AccountRepoState {
+  type AMap = Map[String, Account]
+
+  type Err[A] = String \/ A
+  type ErrState[A] = StateT[Err, AMap, A]
```

Comment:
The name `ErrState` also doesn't look nice. May be `AccountState` or something - error handling is something it does implicitly. So no need to include `Err` as part of the name.